src/app/rundown-view/components/rundown/rundown.component.ts

```diff
@@ -64,9 +93,31 @@ export class RundownComponent implements OnInit, OnDestroy {
 
   public ngOnDestroy(): void {
     this.rundownTimingContextSubscription?.unsubscribe()
+    this.rundownActionsSubscription?.unsubscribe()
   }
 
   public trackSegment(_: number, segment: Segment): string {
     return segment.id
   }
+
+  private updateMiniShelfSegments(): void {
+    this.miniShelfSegments = this.rundown.segments.filter(segment => segment.metadata?.miniShelfVideoClipFile)
+  }
+
+  private updateMiniShelfSegmentActionMappings(): void {
+    this.miniShelfSegmentActionMappings = this.miniShelfSegments.reduce(this.miniShelfSegmentsReducer.bind(this), {})
+  }
+
+  private miniShelfSegmentsReducer(actionMap: Record<string, Tv2VideoClipAction>, segment: Segment): Record<string, Tv2VideoClipAction> {
+    const videoClipFile: string | undefined = segment.metadata?.miniShelfVideoClipFile
+    if (videoClipFile !== undefined) {
+      const action: Tv2VideoClipAction | undefined = this.videoClipActions.find(action => {
+        return action.metadata?.fileName === videoClipFile
+      })
+      if (action !== undefined) {
+        return { ...actionMap, [segment.id]: action }
+      }
+    }
+    return actionMap
```

Comment:
By using "defensive" programming i.e. reversing the if-statement and returning early, you can reduce the layers of nesting.
We should aim for only have one layer of nesting pr method.
This would then become:
```
private miniShelfSegmentsReducer(...): ... {
  const videoClipFile: string | undefined = ...
  if (!videoClipFile) {
    return actionMap
  }
  const action: Tv2VideoClipAction | undefined = ...
  return action ? { ...actionMap, [segment.id]: action } : actionMap
}
```